module/documents/actor/actor.mjs

```diff
@@ -1402,6 +1402,87 @@ export default class Actor5e extends Actor {
 
   /* -------------------------------------------- */
 
+  /**
+   * Get an un-evaluated D20Roll instance used to roll initiative for this Actor.
+   * @param {object} [options]                        Options which modify the roll
+   * @param {D20Roll.ADV_MODE} [options.advantageMode]    A specific advantage mode to apply
+   * @param {string} [options.flavor]                     Special flavor text to apply
+   * @returns {D20Roll}                               The constructed but unevaluated D20Roll
+   */
+  getInitiativeRoll({advantageMode, flavor}={}) {
+
+    // Use a temporarily cached initiative roll
+    if ( this._cachedInitiativeRoll ) return this._cachedInitiativeRoll.clone();
+
+    // Prepare roll data
+    const data = this.getRollData();
+    const flags = this.flags.dnd5e || {};
+    if ( flags.initiativeAdv ) advantageMode ??= dnd5e.dice.D20Roll.ADV_MODE.ADVANTAGE;
+
+    // Standard initiative formula
+    const parts = ["1d20"];
+
+    // Special initiative bonuses
+    if ( "init" in this.system.attributes ) {
+      const init = this.system.attributes.init;
+      parts.push(init.mod);
+      if ( init.prof.term !== "0" ) parts.push(init.prof.term);
+      if ( init.bonus !== 0 ) parts.push(init.bonus);
```

Comment:
I think the way these are typically done in other roll methods is by using data references:

```suggestion
      parts.push("@prof");
      data.prof = init.prof.term;
      parts.push("@bonus");
      data.bonus = init.bonus;
```

Similarly for the `dexCheckBonus` and `globalCheckBonus` below. It's not strictly necessary, but I think it might have some use for 3rd-party consumers. Might also be useful to enforce rules about only adding proficiency bonus once to a given roll.